packages/react-core/src/components/TreeView/TreeViewListItem.tsx

```diff
@@ -94,14 +97,23 @@ const TreeViewListItemBase: React.FunctionComponent<TreeViewListItemProps> = ({
   }, [isExpanded, defaultExpanded]);
 
   const Component = hasCheck ? 'div' : 'button';
```

Comment:
I was able to replicate the behavior from the "with checkboxes" example by updating this to `hasCheck || hasSelectableNodes ? 'div' : 'button'` (more info in another comment below)